elementary/monitor/api/alerts/alerts.py

```diff
@@ -0,0 +1,277 @@
+
+    def get_new_alerts(self, days_back: int, disable_samples: bool = False) -> Alerts:
+        new_test_alerts = self.get_test_alerts(days_back, disable_samples)
+        new_model_alerts = self.get_model_alerts(days_back)
+        new_source_freshness_alerts = self.get_source_freshness_alerts(days_back)
+        return Alerts(
+            tests=new_test_alerts,
+            models=new_model_alerts,
+            source_freshnesses=new_source_freshness_alerts,
+        )
+
+    def get_test_alerts(self, days_back: int, disable_samples: bool = False):
+        pending_test_alerts = self._query_pending_test_alerts(
+            days_back, disable_samples
+        )
+        last_alert_sent_times = self._query_last_test_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_test_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=TestAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_model_alerts(self, days_back: int):
+        pending_model_alerts = self._query_pending_model_alerts(days_back)
+        last_alert_sent_times = self._query_last_model_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_model_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=ModelAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def get_source_freshness_alerts(self, days_back: int):
+        pending_source_freshness_alerts = self._query_pending_source_freshness_alerts(
+            days_back
+        )
+        last_alert_sent_times = self._query_last_source_freshness_alert_times(days_back)
+        alerts_to_skip, alerts_to_send = self._sort_alerts(
+            pending_source_freshness_alerts, last_alert_sent_times
+        )
+        self._skip_alerts(
+            alerts_to_skip=alerts_to_skip, table_name=SourceFreshnessAlert.TABLE_NAME
+        )
+        return alerts_to_send
+
+    def _sort_alerts(
+        self,
+        pending_alerts: AlertsQueryResult[Alert],
+        last_alert_sent_times: Dict[str, str],
+    ) -> Tuple[List[str], AlertsQueryResult[Alert]]:
+        suppressed_alerts = self._get_suppressed_alerts(
+            pending_alerts, last_alert_sent_times
+        )
+        alerts_to_skip = []
+        alerts_to_send = []
+        malformed_alerts_to_send = []
+
+        for alert in pending_alerts.alerts:
+            if alert.id in suppressed_alerts:
+                alerts_to_skip.append(alert.id)
+            else:
+                alerts_to_send.append(alert)
+
+        for alert in pending_alerts.malformed_alerts:
+            if alert.id in suppressed_alerts:
+                alerts_to_skip.append(alert.id)
+            else:
+                malformed_alerts_to_send.append(alert)
+
+        return alerts_to_skip, AlertsQueryResult(
```

Comment:
Shouldn't `alerts_to_skip` be part of `AlertsQueryResult`?
It would create a clearer interface to the function.